src/components/Providers/SelectDatesProvider.jsx

```diff
@@ -1,7 +1,17 @@
 import React, { createContext, useState, useContext, useMemo } from 'react'
 
 export const SelectDatesContext = createContext()
-export const useSelectDatesContext = () => useContext(SelectDatesContext)
+
+export const useSelectDatesContext = () => {
+  const context = useContext(SelectDatesContext)
+
+  if (!context) {
+    throw new Error(
+      'useSelectDatesContext must be used within a SelectDatesProvider'
```

Comment:
in my opinion, the Providers (Trip, SelectDates and Account) should have one test when context is null, checking we throw error.